src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -258,24 +254,24 @@ class InstanceOpFactoryImpl(
         // All of these cases are handled in one way: by creating a new taskId for a resident task based on the previous
         // one. The used function will increment the attempt counter if it exists, of append a 1 to denote the first attempt
         // in version 1.5.
-        val newTaskId = Task.Id.forResidentTask(taskId)
+        val taskIds: Seq[Task.Id] = if (reservedInstance.hasReservation) {
+          val originalIds = if (reservedInstance.tasksMap.nonEmpty) {
```

Comment:
As I mentioned above, this is a little confusing to me. When can we have this case?